### PR TITLE
docs: add config-validation chapter

### PR DESCRIPTION
**Pull request by fgreinacher**
## Changes

Adds top-level documentation for config validation. I think this is worth to be more prominent because it can help folks notice configuration issues earlier. That's especially helpful when Renovate is running on a relaxed schedule, e.g. in self-hosted scenarios.

## Context

Triggered by discussion at https://gitlab.com/renovate-bot/renovate-runner/-/merge_requests/1665#note_1377954509.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

🛠 with ❤️ by [Siemens](https://opensource.siemens.com/)

### CODE REVIEW

File: docs/usage/config-validation.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+title: Config Validation
+description: Validation Renovate configuration files
```
Comment (HonkingGoose) [inline docs/usage/config-validation.md:3]:
```suggestion
description: How to validate Renovate's configuration.
```

I don't know if we need to escape that `'` in the description.

**Review by HonkingGoose — COMMENTED** [review]
I like your idea! The page needs more content though. 🙂

File: docs/usage/config-validation.md (hunk 1 above)
Comment (fgreinacher) [reply docs/usage/config-validation.md:3]: This is OK, it's not ambigious

File: docs/usage/config-validation.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+---
+
+# Config Validation
+
+All [`renovate` distributions](getting-started/running.md#available-distributions) contain a standalone validator program (`renovate-config-validator`) that can be used to validate Renovate's configuration.
+
+The validator program checks files passed as CLI arguments.
+If no argument is given, all [default locations](../configuration-options.md) (if files exist) and the `RENOVATE_CONFIG_FILE` environment variable are checked.
+
+```shell-session
```
Comment (fgreinacher) [inline docs/usage/config-validation.md:13]: Rendered like this:

image

Comment (HonkingGoose) [reply docs/usage/config-validation.md:13]: That's very cool! I did not know we could do that! 🥳

File: docs/usage/config-validation.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+---
```
Comment (HonkingGoose) [inline docs/usage/config-validation.md:1]: We have a "Troubleshooting Renovate" page here: https://docs.renovatebot.com/troubleshooting/

We should insert a link to your new "config validation page" here. 😄

File: docs/usage/config-validation.md (hunk 2 above)
Comment (viceice) [reply docs/usage/config-validation.md:13]: what about using `console` as language keyword?

Comment (HonkingGoose) [reply docs/usage/config-validation.md:13]: Do console and shell-session render the same?

Comment (fgreinacher) [reply docs/usage/config-validation.md:13]: Yes!

File: docs/usage/troubleshooting.md [hunk 4]
```diff
@@ -64,3 +64,9 @@ If none of these steps have helped you, then create a new discussion post to get
 Please locate the relevant parts of the logs as described earlier before asking for help or posting a bug report.
 Do not expect the Renovate maintainers to read through the full logs when trying to help you, as that takes a lot of time on our part.
 If later it turns out that the full logs are necessary, you will be asked for them then.
+
+## Validating configuration changes
+
+Sometimes you will have to change your Renovate configuration to solve a problem.
+The [`renovate-config-validator` program](config-validation.md) helps validate such configuration changes
+without commiting them to your repository.
```
Comment (HonkingGoose) [inline docs/usage/troubleshooting.md:72]:
```suggestion
Sometimes you will have to change your Renovate configuration to solve a problem.
The [`renovate-config-validator` program](config-validation.md) helps validate such configuration changes without commiting them to your repository.
```

File: docs/usage/config-validation.md (hunk 2 above)
Comment (HonkingGoose) [reply docs/usage/config-validation.md:13]: Let's use `console` then, it's shorter and easier to remember.

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

File: docs/usage/config-validation.md (hunk 2 above)
Comment (fgreinacher) [reply docs/usage/config-validation.md]: Sure, done!

**Review by rarkins — APPROVED** [review]

**Pull request merged**